### PR TITLE
Add ASEv3 example and edit the docs.

### PR DESCRIPTION
**Pull request by samlin-msft**
The App Service Environment V3 is GA. In the https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/app_service_environment_v3 example can be updated to support 6 types of ASEv3 deployments. Some attributes are optional for ASEv3 for example: cluster_setting.
 
- ILB ASEv3 (internal_load_balancing_mode = "Web, Publishing" and zone_redundant = false)
- ELB ASEv3 (internal_load_balancing_mode = "None" and zone_redundant = false)
- ILB ASEv3 with AZ support (internal_load_balancing_mode = "Web, Publishing" and zone_redundant = true)
- ELB ASEv3 with AZ support (internal_load_balancing_mode = "None" and zone_redundant = true)
- ILB ASEv3 with dedicated hosts (internal_load_balancing_mode = "Web, Publishing" and dedicated_host_count = 2)
- ELB ASEv3 with dedicated hosts (internal_load_balancing_mode = "None" and dedicated_host_count = 2)

Added example files in examples/app-service-environment-v3 directory as well.

### CODE REVIEW

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @samlin-msft - Thanks for this PR, the /example/ looks great. 

However, I'm afraid the docs file will need to be reverted.  The Arguments and Attributes detailed must be exclusively those used by the resource schema and not variables or other values specified elsewhere in the configuration. Most users will probably have their own modules and external configuration, so need this information. The example in the resource doc should also be as simple as possible to allow users see how a resource is used, without the abstraction of vars.

File: website/docs/r/app_service_environment_v3.html.markdown
Comment (jackofallops) [inline website/docs/r/app_service_environment_v3.html.markdown:114]: Could you leave the ID in, it's always exported on resources.

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @samlin-msft 
Apologies for the delay in returning to this.  I have just one more change of note. If you can fix that up and address the merge conflict, I think this will be good to go.

Thanks again.

File: website/docs/r/app_service_environment_v3.html.markdown
Comment (samlin-msft) [reply website/docs/r/app_service_environment_v3.html.markdown:114]: done!

**Comment by jackofallops** [comment]
Hi @samlin-msft - Could you resolve the merge conflict on this PR?  

For info if you PR from a branch on your fork, rather than `main`, the maintainers can safely assist with that if we need to. 

Thanks!

**Review by mbfrahry — APPROVED** [review]
LGTM after fixing the merge conflict. Thanks for the PR @samlin-msft

**Pull request merged**
